transcoder/src/keyframes.go

```diff
@@ -164,10 +172,46 @@ func getKeyframes(path string, kf *Keyframe) error {
 			ret = ret[:0]
 		}
 	}
+	if len(ret) == 0 {
+		duration, err := getFileDuration(path)
+		if err != nil {
+			return err
+		}
+		ret = make([]float64, 0, int((duration/2)+1))
+		for segmentTime := float64(0); segmentTime < duration; segmentTime += 2 {
+			ret = append(ret, segmentTime)
```

Comment:
You don't need to append, you can use `ret[i] = i *2` if you set the len instead of the cap before.

This is more of a nit, you can keep like this.